install-deps.sh

```diff
@@ -465,6 +465,10 @@ function preload_wheels_for_tox() {
     popd > /dev/null
 }
 
+# install git precommit hooks https://pre-commit.com/
+curl https://pre-commit.com/install-local.py | python -
+pre-commit install > /dev/null
```

Comment:
i'd avoid putting this kind of opinionated setting in `install-deps.sh`, which is supposed to ready the build dependencies and only the build dependencies.

Reply:
i agree, unable to find any place where we can enforce this, added in a separate script